src/test/ui/exclusive-range/exclusive_range_pattern_syntax_collision3.stderr

```diff
@@ -21,11 +21,26 @@ error[E0308]: mismatched types
 LL |     match [5..4, 99..105, 43..44] {
    |           ----------------------- this match expression has type `std::ops::Range<{integer}>`
 LL |         [..9, 99..100, _] => {},
-   |               ^^^^^^^ expected struct `std::ops::Range`, found integer
+   |               ^^  --- this is of type `{integer}`
+   |               |
+   |               expected struct `std::ops::Range`, found integer
+   |
+   = note: expected struct `std::ops::Range<{integer}>`
+                found type `{integer}`
+
+error[E0308]: mismatched types
+  --> $DIR/exclusive_range_pattern_syntax_collision3.rs:5:19
+   |
+LL |     match [5..4, 99..105, 43..44] {
+   |           ----------------------- this match expression has type `std::ops::Range<{integer}>`
+LL |         [..9, 99..100, _] => {},
+   |               --  ^^^ expected struct `std::ops::Range`, found integer
+   |               |
```

Comment:
I still don't understand what's happening here

Reply:
The first error above is for `99` and the other one is for `100`. They now occur twice because the spans are different and so there's no de-duplication. The type error happens because `99..105` is at type `Range<{integer}>` whereas `99..100` is a range pattern at type `{integer}`.

Reply:
In my version I tried changing "this match expression has" to "this match expression requires" in an attempt to clarify these errors

Reply:
It's fine to change that, but that can be in a different PR than this (e.g. yours).

Reply:
The problem I have with this wording is "expected struct `std::ops::Range`, found integer", but now I see that the thing being matched is `[5..4, 99..105, 43..44]`, which accounts for the weirdness.

Reply:
Right, and that part hasn't changed as you can see from the diff. :)